data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/beacon/GetStateWithDataIntegrationTest.java

```diff
@@ -68,16 +68,7 @@ private BeaconState getBeaconStateFromResponse(Response response) throws Excepti
   private tech.pegasys.teku.datastructures.state.BeaconState getInternalState(
       final UnsignedLong slot) {
     try {
-      return combinedChainDataClient
-          .getBlockAndStateInEffectAtSlot(slot)
-          .thenApply(
-              maybeBlockAndState ->
-                  maybeBlockAndState.map(
-                      blockAndState ->
-                          combinedChainDataClient.regenerateBeaconState(
-                              blockAndState.getState(), slot)))
-          .get()
-          .get();
+      return combinedChainDataClient.getStateAtSlotExact(slot).get().get();
```

Comment:
Same as above probably just:
```suggestion
      return combinedChainDataClient.getStateAtSlotExact(slot).join().orElseThrow();
```